server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -762,12 +762,20 @@ public void relocated(
         final String targetAllocationId,
         final Consumer<ReplicationTracker.PrimaryContext> consumer,
         final Consumer<StepListener> performSegRep,
-        final ActionListener<Void> listener
+        final ActionListener<Void> listener,
+        final boolean syncTranslog
     ) throws IllegalIndexShardStateException, IllegalStateException, InterruptedException {
         assert shardRouting.primary() : "only primaries can be marked as relocated: " + shardRouting;
         try (Releasable forceRefreshes = refreshListeners.forceRefreshes()) {
             indexShardOperationPermits.blockOperations(30, TimeUnit.MINUTES, () -> {
                 forceRefreshes.close();
+
+                // Since all the index permits are acquired at this point, the translog buffer will not change.
+                // It is safe to perform sync of translogs now as this will ensure for remote-backed indexes, the
+                // translogs has been uploaded to the remote store.
+                if (syncTranslog) {
+                    maybeSync();
```

Comment:
If we don't perform sync here for indexes with remote translog enabled, is there a chance of data loss?

Reply:
Yes, it is possible that data loss can happen if `index.translog.durability` is set as `async`. When `async` the translog sync happens every `index.translog.sync_interval`. If the sync interval is too higher, then the shard can get closed before the sync happens when the relocation has completed.